server/service/externalsvc/zendesk.go

```diff
@@ -0,0 +1,155 @@
+package externalsvc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+	"github.com/nukosuke/go-zendesk/zendesk"
+)
+
+// Zendesk is a Zendesk client to be used to make requests to the Zendesk external service.
+type Zendesk struct {
+	client  *zendesk.Client
+	groupID int64
+}
+
+// ZendeskOptions defines the options to configure a Zendesk client.
+type ZendeskOptions struct {
+	URL      string
+	Email    string
+	APIToken string
+	GroupID  int64
+}
+
+// NewZendeskClient returns a Zendesk client to use to make requests to the Zendesk external service.
+func NewZendeskClient(opts *ZendeskOptions) (*Zendesk, error) {
+	if os.Getenv("TEST_ZENDESK_CLIENT") == "true" {
+		return NewZendeskTestClient(opts)
+	}
+	client, err := zendesk.NewClient(fleethttp.NewClient())
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := url.Parse(opts.URL)
+	if err != nil {
+		return nil, err
+	}
+	subparts := strings.Split(url.Host, ".")
+	subdomain := subparts[0]
+
+	client.SetSubdomain(subdomain)
```

Comment:
Out of curiousity, why not use `client.SetEndpointURL`?

Reply:
It is recommended only for testing per the comments in the third-part package: 
// SetEndpointURL replace full URL of endpoint without subdomain validation.
// This is mainly used for testing to point to mock API server.